### ISSUE TITLE
Detected blocking call warnings  in 2023.7.0b2

### ISSUE THREAD

**Issue by dummylabs** [dummylabs/thewatchman#137]
New beta version of HA gives warning messages in the log again. Previous warnings were fixed in #135, those are introduced in 2023.7:

```
2024-06-27 06:01:55.040 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to iglob with args ('/config/**/*.yaml',) inside the event loop by custom integration 'watchman' at custom_components/watchman/utils.py, line 160: for filename in glob.iglob(folder, recursive=True): (offender: /workspaces/thewatchman/custom_components/watchman/utils.py, line 160: for filename in glob.iglob(folder, recursive=True):), please create a bug report at https://github.com/dummylabs/thewatchman/issues
For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#iglob
```

```
2024-06-27 06:01:55.044 WARNING (MainThread) [homeassistant.util.loop] Detected blocking call to scandir with args ('/config',) inside the event loop by custom integration 'watchman' at custom_components/watchman/utils.py, line 160: for filename in glob.iglob(folder, recursive=True): (offender: /usr/local/lib/python3.12/glob.py, line 160: with os.scandir(arg) as it:), please create a bug report at https://github.com/dummylabs/thewatchman/issues
For developers, please see https://developers.home-assistant.io/docs/asyncio_blocking_operations/#scandir
```

**Comment by dummylabs** [comment]
Fixed code is in the main branch, will be added to the next release. Meanwhile it is available in HACS (Redownload->Show beta versions->main).

**Comment by Mariusthvdb** [comment]
ws just about to open an issue, and you beat us to it. cool!
thanks for being fast!

**Comment by johntdyer** [comment]
So not sure why but it wont allow me to download main any more, it keeps defaulting back to 0.6.2...

**Comment by Mariusthvdb** [comment]
Yes that is a Hacs issue I filed already at hacs

dev said it will be fixed in new update. For now, just download manually

**Comment by dummylabs** [comment]
Fixed in 0.6.3
